### PR TITLE
Unit tests for photon pusher

### PR DESCRIPTION
**Pull request by lucafedeli88**
I've added a test which should check if the photon pusher is working properly. 

Photons should keep their initial momentum and should move along straight lines at the speed of light. 
The `inputs` file initializes 16 photon particles with momenta having different directions and magnitudes. Photon particles are then evolved for 50 steps. The python script `check.py` verifies that the final position is close enough to the expected one (tolerance of ~ 1e-14 on the relative error) and that the final momentum is **exactly** equal to the expected one. 

Btw, where does m_e = 9.10938291e-31 (used in WarpX) come from? It is slightly different from the value currently reported by Wikipedia.

### CODE REVIEW

**Comment by lucafedeli88** [comment]
I've just put [WIP] because I've realized that I have to pass the diag file name to the check.py script, but I don't know how to do that.

File: Examples/Tests/SingleParticle/inputs [hunk 1]
```diff
@@ -1,23 +1,137 @@
-max_step = 1
```
Comment (MaxThevenet) [inline Examples/Tests/SingleParticle/inputs]: This file is used in another test (bilinear_filter). Modifying it broke the other test.

Comment (lucafedeli88) [reply Examples/Tests/SingleParticle/inputs]: Sorry, modifying this file was a mistake. I've just reverted it to its previous version

**Comment by lucafedeli88** [comment]
> It seems that you accidentally modified  Examples/Tests/SingleParticle/inputs. Could you revert to previous version of this file? 

Yes, it was a mistake. I've reverted the file to its previous version.

>The test that you propose covers many situations, which is nice, but  since the interface of the code is >changing regularly, it may be  cumbersome to update the test periodically due to the large number of  >species. Maybe we could use a test that initializes a distribution of  particles, and check global quantities >(there are a number of options  here) on this distribution, as done in rigid_injection tests for  instance. >That is just a suggestion, I can help build a test if you  want. 

Would writing a python script to generate the inputfile and the check.py file help? If the interface changes, it would be enough to update this script and run it to generate the test. In some ways, this is similar to what is already done for the parser of the inputfile.

>Note that, without doing anything more, this test will also be  part of the regression test suit, that checks >agreement with a benchmark  every night. If a tiny bug were to modify the propagation of each  photon >without modifying the total distribution too much, the python  script might not catch it but the regression >tests will. 

Ok. I didn't know that. Would it be possible to exclude the test from the regression test suite? How does the agreement with the benchmark is checked? Would increasing tolerance a little bit help? 
When you say that a tiny bug might modify the propagation of each photon, do you mean that the final position could be slightly different? This would be ok I think (unless the error is too big). However, the photon momentum is not supposed to change. It would be better though to read the initial momentum from the first output file and compare it with the last one. 

>As for your question about how to pass a plotfile, the last plotfile  is automatically passed as argument to >this script, and read by filename = sys.argv[1] at the beginning of your file.

Ok. Then it's already doing what it should do.

**Comment by lucafedeli88** [comment]
I've just pushed a new version of check.py script. If it runs without arguments, it regenerates the inputfile from scratch (from a list of species names and momenta). In this way the inputfile should be easier to maintain. If something in the interface changes, it will be enough to update the check.py script and re-run it.

**Pull request merged**